### DISCUSSION TITLE
Hyperlinks for phpstorm

### DISCUSSION THREAD

**Discussion by Eitan-brightleaf** [dandavison/delta#1933] [Q&A]
Hi, I'm new to Delta and it looks great. I'm trying to get it to open up the file to the right place in php storm though and can't seem to manage it. I have phpstorm installed through jetbrains which based on this [link ](https://youtrack.jetbrains.com/issue/TBX-6055/open-phpStorm-file-via-an-url) changes things a bit. the closest i got was `hyperlinks-file-link-format = "jetbrains://php-storm/navigate/reference?project=<PROJECT_NAME>&path={path}&line={line}"` which opened up the right project but not any files.

As a side note, since in the meantime I am using the documented `hyperlinks-file-link-format = "vscode://file/{path}:{line}"` for vs code is there any way to open the file with the whole folder?

**Comment by th1000s** [comment]
Does the `jetbrains://` protocol expect a relative path vs. delta's absolute one? I also see `project=..` there from where it might start looking. Because delta and other external programs can't know the project name it would be better if the Jebrains IDEs can translate absolute paths, might need a feature request.

**Reply by Eitan-brightleaf** [reply]
I have no idea.
Do you mean a feature request for Jetbrains or Delta?
The [online documentation](https://dandavison.github.io/delta/hyperlinks.html) for Delta says that it should work for Jetbrains IDEs but I'm not sure if that was older versins, or if they aren't installed through the toolbox or maybe I'm just doing something wrong...

**Reply by th1000s** [reply]
I see the protocol there uses a much simpler format, is this registered with your OS and does it work `phpstorm://open?file={path}&line={line}`? (or maybe first [aik099/PhpStormProtocol](https://github.com/aik099/PhpStormProtocol) is needed?)

**Reply by Eitan-brightleaf** [reply]
The suggested format does not work. My OS (windows) doesn't recognize it. I could install that if I need to but the readme there has a link to comments which claim that the url protocol works. But hard to follow from the thread what exactly does and doesn't, especially as it goes back over a decade. I'll open a ticket with them also and see what they say and report it here if no solution is found first.
